components/cosmetic_filters/resources/data/content_cosmetic.ts

```diff
@@ -35,10 +23,39 @@ let notYetQueriedClasses: string[]
 let notYetQueriedIds: string[]
 let cosmeticObserver: MutationObserver | undefined = undefined
 
-const allSelectorsToRules = new Map<string, number>()
-const cosmeticStyleSheet = new CSSStyleSheet()
-
-const injectScriptlet = (text: string) => {
+window.content_cosmetic = window.content_cosmetic || {}
```

Comment:
we do call the cosmetic_filters.ts several times, we would need to call that whole script again to make that function call that you suggested, as it would live in the same ts. I would propose to create a follow up to clean that up and make a solution with moving that JS functions inside the ts to have all pluses ts brings to us with types checks and etc.

Reply:
In that case, please could you record somewhere the times that the script gets injected multiple times (perhaps comment in the script but if not then at least c++ or even this PR) so that it's obvious when changing this script that it has to account for that? I'd be happy to help as a follow-up to clean this up, as long as we will remember because I think it will be fragile and we won't get much false negative or error notification.

Reply:
I put a note at the beginning of the script that it is called multiple times
`// That script is executed from
// brave/browser/brave_cosmetic_resources_tab_helper.cc several times:
// - for scriptlets injection;
// - for cosmetic filters work with CSS and stylesheet. That work itself
//   could call the script several times.`